Replace the Find out More form on the /districts page with the new Hubspot one

TurnerRiley opened the pull request:
Replaces the Find out More form on the /districts page with the new Hubspot one. This required several extra files to be changed so that we could still log when the form was submitted. I also updated them to match our general format of having `.haml` views with associated `.js` files rather than the `.html` file currently being used.

As for the id's necessary to set up the form, I created secrets for them in AWS. This is my first time making secrets so I wasn't sure exactly which ones to create them for, so I made the following to ensure the form would show up on staging, test, levelbuilder, and production (but am open to pushback to creating/deleting any!):
`hubspot_portal_id`: [staging](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=staging%2Fcdo%2Fhubspot_portal_id&region=us-east-1), [test](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=test%2Fcdo%2Fhubspot_portal_id&region=us-east-1), [levelbuilder](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=levelbuilder%2Fcdo%2Fhubspot_portal_id&region=us-east-1), [production](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=production%2Fcdo%2Fhubspot_portal_id&region=us-east-1)
`hubspot_admin_interest_form_id`: [staging](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=staging%2Fcdo%2Fhubspot_admin_interest_form_id&region=us-east-1), [test](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=test%2Fcdo%2Fhubspot_admin_interest_form_id&region=us-east-1), [levelbuilder](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=levelbuilder%2Fcdo%2Fhubspot_admin_interest_form_id&region=us-east-1), [production](https://us-east-1.console.aws.amazon.com/secretsmanager/secret?name=production%2Fcdo%2Fhubspot_admin_interest_form_id&region=us-east-1)

## Previous form
![old form](https://github.com/code-dot-org/code-dot-org/assets/56283563/259cfa7a-84fb-4b08-ab78-a39fd7287eaa)

## New form with demo (including logging)
https://github.com/code-dot-org/code-dot-org/assets/56283563/82e98eb5-11a8-478f-a5ee-ff60044e3281

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1886)

## Testing story
Local testing.

## Security

kelbyhawn reviewed (commented):
Can we customize the styling of the Hubspot form to keep it consistent with our brand?

TurnerRiley commented:
> Can we customize the styling of the Hubspot form to keep it consistent with our brand?

It looks like if we have a Marketing Hub Starter, Professional, or Enterprise subscription, we can edit the styling on the website, or if we have a Marketing Hub or Content Hub Professional or Enterprise subscription with our HubSpot account then we can get the HTML of the form instead and then use that rather than the current script tag (allowing us to use our own stylesheets). @dmcavoy What type of subscription do we have and, if we have one that supports it, should I create a ticket for investigating styling?

dmcavoy commented:
> > Can we customize the styling of the Hubspot form to keep it consistent with our brand?
> 
> From the way it's currently incorporated, there is not an easy way to update the styles. [It looks like](https://knowledge.hubspot.com/forms/how-can-i-share-a-hubspot-form-if-im-using-an-external-site) if we have a Marketing Hub Starter, Professional, or Enterprise subscription, we can edit the styling on the HubSpot website, or if we have a Marketing Hub or Content Hub Professional or Enterprise subscription with our HubSpot account then we can get the HTML of the form instead and then use that rather than the current script tag (allowing us to use our own stylesheets). @dmcavoy What type of subscription do we have and, if we have one that supports it, should I create a ticket for investigating styling?

@TurnerRiley I don't know. I would reach out to Nikki N. to get details and see if we can update. I think we can ship it as is and then follow up on that as needed

kelbyhawn reviewed (approved):
Thanks for including the video! I'm not sure about the secrets, so someone else might want to take a look at that, but approving based on everything else (and 🤞 we can update the styles in a future PR!)

bethanyaconnor commented on pegasus/sites.v3/code.org/views/admins_email_signup_form.haml:
```diff
@@ -0,0 +1,14 @@
+%script{src: webpack_asset_path('js/code.org/views/admins_email_signup_form.js')}
+%script{charset: "utf-8", type: "text/javascript", src: "//js.hsforms.net/forms/embed/v2.js"}
+
+:javascript
```
just out of curiousity, why can't this also be moved to the JS file?

TurnerRiley replied:
Oh true! I had refactored the file from using an `html` script tag to `haml` but didn't check a step further of moving that over to the new `js` file as well. I'll get on that!

TurnerRiley replied:
Oh actually I'm realizing I'm not sure I'll be able to import `hbspt` into the javascript file that's being retrieved in line 2

bethanyaconnor reviewed (approved):
Thanks for the refactor! One question but LGTM!

TurnerRiley commented:
> > > Can we customize the styling of the Hubspot form to keep it consistent with our brand?
> > 
> > 
> > From the way it's currently incorporated, there is not an easy way to update the styles. [It looks like](https://knowledge.hubspot.com/forms/how-can-i-share-a-hubspot-form-if-im-using-an-external-site) if we have a Marketing Hub Starter, Professional, or Enterprise subscription, we can edit the styling on the HubSpot website, or if we have a Marketing Hub or Content Hub Professional or Enterprise subscription with our HubSpot account then we can get the HTML of the form instead and then use that rather than the current script tag (allowing us to use our own stylesheets). @dmcavoy What type of subscription do we have and, if we have one that supports it, should I create a ticket for investigating styling?
> 
> @TurnerRiley I don't know. I would reach out to Nikki N. to get details and see if we can update. I think we can ship it as is and then follow up on that as needed

To circle back. Nikki and I were able to update the text field to have small rounded corners, made the font of the fields a sans-serif font similar to Metropolis (it wasn't available unfortunately), updated the font sizes and colors to match that of the previous form, and updated the button color and text to match that of the previous form. There will be some small stylistic differences but it's as close as we can get based on our current HubSpot plan.
